src/singer_clojure/messages.clj

```diff
@@ -0,0 +1,83 @@
+(ns singer-clojure.messages
+  (:require [clojure.data.json :as json]
+            [clojure.test :refer [is]]))
+
+(defn valid?
+  "Function to evalulate whether a singer message is valid"
+  [message]
+  (and (#{"SCHEMA" "STATE" "RECORD" "ACTIVATE_VERSION"} (message "type"))
+       (case (message "type")
+         "SCHEMA"
+         (message "schema")
+
+         "STATE"
+         (message "value")
+
+         "RECORD"
+         (message "record")
+
+         "ACTIVATE_VERSION"
+         (message "version"))))
+
+(defn write!
+  "Writes a valid singer message to stdout"
+  [message]
+  {:pre [(is (valid? message) (format "message %s is not valid" message))]}
```

Comment:
Do we need this `is`? If so, then should we be consistent and make the other pre check use `is` and have a meaningful message?

Reply:
The other place being `singer_clojure.messages/deserialize-stream`

Reply:
Maybe we need to make `singer_clojure.messages/deserialize-stream` because `singer_clojure.parse/state` also uses `is`

Reply:
Using is here was a way to create a message when the condition failed. There was a stackoverflow post that suggested this: https://stackoverflow.com/questions/17390962/can-i-include-a-message-in-pre-and-postconditions

But I agree that being consistent and using this for all pre/post conditions should probably be done. We only added it when we saw one failed but it provided no information to debug why it failed.

Reply:
Fixed, will update PR shortly